csharp/src/Ice/ObjectAdapter.cs

```diff
@@ -1406,57 +1388,53 @@ private List<EndpointI> parseEndpoints(string endpts, bool oaEndpoints)
 
     private EndpointI[] computePublishedEndpoints()
     {
-        List<EndpointI> endpoints;
+        IEnumerable<EndpointI> endpoints;
         if (_routerInfo is not null)
         {
-            //
             // Get the router's server proxy endpoints and use them as the published endpoints.
-            //
-            endpoints = new List<EndpointI>();
-            foreach (EndpointI endpt in _routerInfo.getServerEndpoints())
-            {
-                if (!endpoints.Contains(endpt))
-                {
-                    endpoints.Add(endpt);
-                }
-            }
+            endpoints = _routerInfo.getServerEndpoints().Distinct();
         }
         else
         {
-            //
-            // Parse published endpoints. If set, these are used in proxies
-            // instead of the connection factory endpoints.
-            //
-            string endpts = _instance.initializationData().properties!.getProperty(_name + ".PublishedEndpoints");
+            // Parse published endpoints. If set, these are used instead of the connection factory endpoints.
+            string endpts = _instance.initializationData().properties!.getProperty($"{_name}.PublishedEndpoints");
```

Comment:
I would remove this variable and move this to the `parseEndpoints` call

```
endpoints = parseEndpoints(
    _instance.initializationData().properties!.getProperty($"{_name}.PublishedEndpoints"),
    oaEndpoints: false);
```

Reply:
Fixed as suggested.